[all] Remove SofaSimulationTree

damienmarchal opened the pull request:
As we have DAGNode there is no need to keep something that do less :)

This is connected discussion in PR #1642


______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

damienmarchal committed:
[all] Remove SofaSimulationTree

As we have DAGNode there is no need to keep something that do less :)

damienmarchal commented on SofaKernel/modules/SofaHelper/src/sofa/helper/system/FileRepository.cpp:
```diff
@@ -238,7 +238,7 @@ bool FileRepository::findFileIn(std::string& filename, const std::string& path)
     }
-    catch (std::exception e)
+    catch (std::exception& e)
     {
```
Yes, removes a warning at compilation time.

damienmarchal commented on SofaKernel/modules/SofaSimulationCore/src/sofa/simulation/ExportDotVisitor.cpp:
```diff
@@ -0,0 +1,360 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
```
Yes it is related as it was in SofaSimulationTree so I move it close to other exporters (the other alternative would have been to removed it).

damienmarchal commented on SofaKernel/modules/SofaDefaultType/src/sofa/defaulttype/TypeInfoID.cpp:
```diff
@@ -19,7 +19,6 @@
 *                                                                             *
 * Contact information: contact@sofa-framework.org                             *
 ******************************************************************************/
-#pragma once
```
cpp files are not supposed to be included so why having a #pragma once in a .cpp ?

damienmarchal commented on SofaKernel/modules/SofaSimpleFem/src/SofaSimpleFem/TetrahedronDiffusionFEMForceField.inl:
```diff
@@ -30,7 +30,11 @@
+#include <iostream>
+#include <fstream>
+#include <cstring>
+
```
Hi hugo, thanks for the carefull review, I just checked back the compilation after removing the lines and they don't seems needed.

damienmarchal commented on applications/plugins/Registration/Registration_run/oneTetra.h:
```diff
@@ -24,7 +24,7 @@
 #include <SofaBaseVisual/VisualStyle.h>
-
+#include <SofaBaseTopology/MeshTopology.h>
 #include <sofa/core/objectmodel/Context.h>
```
MeshTopology is used in oneTetra.h 
Line 38: using sofa::component::topology::MeshTopology and others... 
It was probably previously included under the hood but the #include in #include

damienmarchal committed:
[SofaGeneral] Fix compilation issue.

damienmarchal committed:
[SofaGeneralImplicitOde] FIX invalid find_package(SofaSimulationCommon)

A SofaSimulationCommon is a subpackage of SofaSimulation...
SofaSimulation is the one to take.

damienmarchal committed:
[SofaGeneralExplicitOdeSolver] FIX invalid find_package(SofaSimulationCommon)

A SofaSimulationCommon is a subpackage of SofaSimulation...
SofaSimulation is the one to take.

damienmarchal committed:
[SofaTest] Add missing include to DAGNode.h in BroadPhase_test.h

damienmarchal committed:
[SofaGeneralLoader] Fix the missing SofaSimulationCore in the link targets.

damienmarchal committed:
[SofaMiscSolver] Add missing package SofaSimulation in CMakeLists.txt

Signed-off-by: Damien Marchal <damien.marchal@univ-lille1.fr>

damienmarchal committed:
[SofaPhysicsAPI] Replace SofaTreeSimulation with SofaGraphSimulation

damienmarchal committed:
[SofaSimulationTree] Add transitional package to warn about removal of SofaSImulatioNTree

damienmarchal committed:
[Modeler] Update few location that were using SofaSimulationTree

damienmarchal committed:
[SofaKernel] Remove deprecated SofaSimulationTree in SofaSImulation/CmakeLists.txt

damienmarchal committed:
[SofaSphFluid] remove link to tbb that was introduced by mistake.

damienmarchal committed:
[tutorials] Use SofaSimulationGraph/simpleapi to create a simulation instead of accessing DAGSimulation.h

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into pr-remove-simulationtree

# Conflicts:
#	SofaKernel/modules/SofaDefaultType/src/sofa/defaulttype/TypeInfoID.cpp
#	SofaKernel/modules/SofaHelper/src/sofa/helper/system/FileRepository.cpp
#	SofaKernel/modules/SofaSimulationTree/GNode.cpp
#	SofaKernel/modules/SofaSimulationTree/init.cpp

damienmarchal commented:
[ci-build][with-scene-tests][with-regression-tests]

damienmarchal committed:
[SofaKernel] Cosmetic removal of an un-needed end of line in PluginManager.cpp

damienmarchal committed:
[modules] Cosmetic removal of duplicated text in BaseGUI.cpp & GUIManager.cpp

damienmarchal committed:
[caduceus.scn] Add missing Requires plugin & put them in a dedicated node.

Because of pluginization we are going to have more and more plugins which
can reduce the scenegraph visibility. Putting them in a dedicated Node with a clear
name (eg: RequiredPlugins, Settings, RuntimeSettings) helps in readbility.

fredroy commented on SofaKernel/modules/SofaSimulationTree/ExportDotVisitor.h:
```diff
@@ -19,93 +19,4 @@
-} // namespace sofa
-
-#endif
+#error "SofaSimulationTree was removed at release 21.06. You need to update your code by replacing #include<SofaSimulationTree/ExportDotVisitor.h> with #include<sofa/simulation/ExportDotVisitor.h>
```
No end of string? (no double quotes at the end)
(same for the other #error message)

damienmarchal replied:
Good eyes, 
as there is no need for quotes at all (https://en.cppreference.com/w/c/preprocessor/error) I will remove them. 

EDIT: Done.

damienmarchal committed:
[all] Unify (i.e remove) the use of double quote in the #error directive

The standard speecify there is no need for quotes/double quotes
https://en.cppreference.com/w/c/preprocessor/error

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into pr-remove-simulationtree
